src/transformStateWithClones.js

```diff
@@ -6,8 +6,27 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let stateCopy = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      stateCopy = {};
+    } else if (action.type === 'addProperties') {
+      stateCopy = { ...stateCopy, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      stateCopy = { ...stateCopy };
```

Comment:
This line is not needed, won't change your code and results, but is adding a unnecessary computation process